src/ln/channelmonitor.rs

```diff
@@ -286,12 +286,27 @@ impl ChannelMonitor {
 				}
 			}
 		}
-		// TODO: Prune payment_preimages no longer needed by the revocation (just have to check
-		// that non-revoked remote commitment tx(n) do not need it, and our latest local commitment
-		// tx does not need it.
+
+		let mut pruned_payment_preimages = HashMap::with_capacity(self.payment_preimages.capacity());
```

Comment:
Can you use self.payment_preimages.retain instead of creating an ew map and drain()ing the old one? Should be a bit less overhead to doing it that way.

Reply:
Should be fine